bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2923,23 +2923,36 @@ private static boolean hasFillerValue(Type type, List<Type> unanalyzedTypes) {
             case TypeTags.ANY_TAG:
                 return true;
             case TypeTags.ARRAY_TAG:
-                return checkFillerValue((BArrayType) type);
+                return checkFillerValue((BArrayType) type, unanalyzedTypes);
             case TypeTags.FINITE_TYPE_TAG:
                 return checkFillerValue((BFiniteType) type);
             case TypeTags.OBJECT_TYPE_TAG:
                 return checkFillerValue((BObjectType) type);
             case TypeTags.RECORD_TYPE_TAG:
                 return checkFillerValue((BRecordType) type, unanalyzedTypes);
             case TypeTags.TUPLE_TAG:
-                BTupleType tupleType = (BTupleType) type;
-                return tupleType.getTupleTypes().stream().allMatch(TypeChecker::hasFillerValue);
+                return checkFillerValue((BTupleType) type, unanalyzedTypes);
             case TypeTags.UNION_TAG:
                 return checkFillerValue((BUnionType) type, unanalyzedTypes);
             default:
                 return false;
         }
     }
 
+    private static boolean checkFillerValue(BTupleType tupleType,  List<Type> unAnalyzedTypes) {
+        if (unAnalyzedTypes.contains(tupleType)) {
+            return true;
+        }
+        unAnalyzedTypes.add(tupleType);
+
+        for (Type member : tupleType.getTupleTypes()) {
+            if (!hasFillerValue(member, unAnalyzedTypes)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Can you verify if the codecov comments are false negatives?

Reply:
The flow goes through the 'true' scenario for examples under `testFillerValueCyclicTuple` in  `type-definitions-cyclic-negative`